subprojects/platform-play/src/main/java/org/gradle/language/twirl/internal/DefaultTwirlSourceSet.java

```diff
@@ -32,6 +32,7 @@
     private TwirlImports defaultImports = TwirlImports.SCALA;
     private List<TwirlTemplateFormat> userTemplateFormats = Lists.newArrayList();
     private List<String> additionalImports = Lists.newArrayList();
+    private List<String> constructorAnnotations = Lists.newArrayList();
```

Comment:
As a `ListProperty`, this will need to be created via a `ObjectFactory`.  Since this class doesn't have one of those yet, you'll have to add a constructor with `@Inject` and inject a `ObjectFactory` instance. I think this should just work.

Reply:
The class `DefaultTwirlSourceSet` inherits from `BaseLanguageSourceSet` that has the property `objectFactory` to be used its subclasses. So it's not necessary to have the constructor 
After changed the property to `ListProperty` must of the integration tests are failing due to `java.io.NotSerializableException: org.gradle.api.internal.provider.DefaultListProperty`. Do you have any idea if this objects will become `Serializable`?